src/views/collections/CreateCollection/EditCollection.vue

```diff
@@ -0,0 +1,218 @@
+    Prop,
+  } from 'vue-property-decorator';
+  import { RouteCallback, Route } from 'vue-router';
+  import axios, { AxiosInstance } from 'axios';
+  // @ts-ignore
+  import EternaPage from '@/components/PageLayout/EternaPage.vue';
+  import TagsPanel from '@/components/Sidebar/TagsPanel.vue';
+  import Utils from '@/utils/utils';
+  import { CollectionItem, PuzzleItem } from '@/types/common-types';
+  import CollectionInfo from './components/CollectionInfo.vue';
+  import CollectionPuzzles from './components/CollectionPuzzles.vue';
+
+  @Component({
+    components: {
+      EternaPage,
+      TagsPanel,
+      CollectionInfo,
+      CollectionPuzzles,
+    },
+  })
+  export default class EditCollection extends Vue {
+    @Prop({ required: true }) private loading!: boolean;
+
+    private title = '';
+
+    private desc = '';
+
+    private puzzlenames: PuzzleItem[] = [];
+
+    private puzzlelist: PuzzleItem[] = [];
+
+    private picture: string = '';
+
+    private newPicture: File | null = null;
+
+    private data: boolean = false;
+
+    fetchData: () => Promise<void> | undefined = async () => {};
+
+    mounted() {
+      this.fetch();
+    }
+
+    async fetch() {
+      const collection = (
+        await this.$http.get(`/get/?type=collection&nid=${this.$route.params.id}`)
+      ).data.data.collection as CollectionItem;
+      if (collection.puzzles != null) {
+        const puzzlelist = collection.puzzles.split(',');
+        Object.values(puzzlelist).forEach(async (puzz) =>
+          this.puzzlelist.push(
+            (await (
+              await this.$http.get(`/get/?type=puzzle&nid=${parseInt(puzz, 10)}`)
+            ).data.data.puzzle) as PuzzleItem,
+          ),
+        );
+      }
+      this.title = collection.title;
+      this.desc = collection.desc;
+      this.picture = collection.image;
+    }
+
+    async submit() {
+      this.loading = true;
+      const data = new FormData();
+      data.set('collection-title-input', this.title);
+      data.set('nid', this.$route.params.id);
+      data.set(
+        'collection-description-input',
+        this.desc
+      );
+      const puzzleids: String[] = [];
+      this.puzzlelist.forEach((e) => puzzleids.push(e.id));
+      data.set('collection-puzzles', puzzleids.toString());
+      if (this.newPicture) data.append(`files[picture_upload]`, this.newPicture);
+      data.set('type', 'edit_collection');
+
+      try {
+        const res = await this.$http.post('/post/', data, {
+          headers: {
+            'Content-type': 'multipart/form-data',
+          },
+        });
+        this.loading = false;
+        const error = res?.data?.data?.error;
+        if (error) throw new Error(error);
+        this.$router.push(`/collections/${this.$route.params.id}`);
+      } catch (e: any) {
+        const r = this.$notify({
+          type: 'error',
+          title: 'Error',
+          text: e.message,
+        });
+      }
+    }
+
+    async deleteCollection() {
```

Comment:
Do we want to have some sort of confirmation dialog? Especially since the delete button is right next to the update button